zuul-core/src/main/java/com/netflix/zuul/message/http/HttpRequestMessageImpl.java

```diff
@@ -528,6 +533,16 @@ public int getOriginalPort()
         return port;
     }
 
+    @Override
+    public Optional<Integer> getClientDestinationPort() {
```

Comment:
Don't think `Optional` is supposed to be used as a generic return type.

Reply:
Not sure I understand this. `Optional` is not recommended as a parameter type
or in substitution for empty collections. I'm not aware of reasons for not using it for response types.

Reply:
Yeah, just an opinion, feel free to ignore.